Mathlib/Topology/Homeomorph.lean

```diff
@@ -701,6 +701,21 @@ def ulift.{u, v} {X : Type u} [TopologicalSpace X] : ULift.{v, u} X ≃ₜ X whe
   continuous_invFun := continuous_uLift_up
   toEquiv := Equiv.ulift
 
+/-- `Equiv.sumArrowEquivProdArrow` as a homeomorphism. The natural homeomoorphism
+`(ι ⊕ ι' → X) ≃ₜ (ι → X) × (ι' → X)` -/
+def sumArrowEquivProdArrow {ι ι' : Type*} : (ι ⊕ ι' → X) ≃ₜ (ι → X) × (ι' → X)  where
+  toEquiv := Equiv.sumArrowEquivProdArrow _ _ _
+  continuous_toFun := by
+    simp only [Equiv.sumArrowEquivProdArrow, Equiv.coe_fn_mk, continuous_prod_mk]
+    continuity
+  continuous_invFun := continuous_pi fun i ↦ match i with
+    | .inl i => by apply (continuous_apply _).comp' continuous_fst
+    | .inr i => by apply (continuous_apply _).comp' continuous_snd
+
+/-- The natural homeomorphism between `(Fin m → X) × (Fin n → X)` and `(Fin (m + n) → X)`.-/
+def finArrowProdHomeomorphFinAddArrow (m n : ℕ) : (Fin m → X) × (Fin n → X) ≃ₜ (Fin (m + n) → X) :=
```

Comment:
Could this be
```suggestion
def _root_.Fin.appendHomeomorph (m n : ℕ) : (Fin m → X) × (Fin n → X) ≃ₜ (Fin (m + n) → X) :=
```
and can you please show `appendHomeomorph.toEquiv = appendEquiv`?

Reply:
Would it make sense to have a lemma stating that before and then define `Fin.appendHomeomorph` using `Fin.appendEquiv` so that they could be definitionally equal?

Reply:
Sorry, I'm not sure I understand your suggestion. How can you prove `appendHomeomorph.toEquiv = appendEquiv` before `appendHomeomorph` is defined?

Reply:
I mean showing that `appendEquiv` is equal to the composition that I right now define `appendHomeomorph` to be.

Reply:
> and then define `Fin.appendHomeomorph` using `Fin.appendEquiv` so that they could be definitionally equal?

I think this is a good idea, though it might be sensible to add the basic `Continuous fun (x, y) => Fin.append x y` as a lemma too.

Reply:
Maybe. You can try! If you do so, I suggest making that lemma `private`